File: client/src/app/modules/quotation/services/service.family.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { ServiceFamily } from '../../quotation/model/ServiceFamily';

@Injectable({
  providedIn: 'root'
})
export class ServiceFamilyService extends AppRestService<ServiceFamily>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getServiceFamilies() {
    return this.getList(new HttpParams(), "service-families");
  }
  
   addOrUpdateServiceFamily(serviceFamily: ServiceFamily) {
    return this.addOrUpdate(new HttpParams(), "service-family", serviceFamily, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/quotation/service/ServiceFamilyService.java
package com.jss.osiris.modules.quotation.service;

import java.util.List;

import com.jss.osiris.modules.quotation.model.ServiceFamily;

public interface ServiceFamilyService {
    public List<ServiceFamily> getServiceFamilies();

    public ServiceFamily getServiceFamily(Integer id);
	
	 public ServiceFamily addOrUpdateServiceFamily(ServiceFamily serviceFamily);
}
